docs/autogen/dest.py

```diff
@@ -105,6 +114,43 @@ def remove_except(path, keep):
                 shutil.rmtree(item)
 
 
+def gen_versions(path: Path, current: str):
+    """Returns a dict compatible with the Pydata versions-format"""
+
+    def version_key(name):
+        match = re.match(r"v(\d+)\.(\d+)\.(\d+)", name)
+        if match:
+            return (1, -int(match.group(1)), -int(match.group(2)), -int(match.group(3)))
+        return (0, name)
```

Comment:
The return type of this is weird, either you return a tuple with four numbers or a number and a string.
Additionally, negating the numbers is unclear. I would prefer if the call to `sorted()` had `reverse=True` instead.

Reply:
It is done to define an ordering that is not just alphanumerical. I want to separate branch names and version tags. Branch-names comes before version-tags and version-tags are in descending order.

I do prefer to have this logic inside the key-function, since that is what it is there for, to define ordering. By adding ``reverse=True`` then that would affect the ordering of the branch names as well, which it should not.

Reply:
I have added a doc-string describing how the function sorts, which might help with the non-obviousness of the code.